avidemux/qt4/ADM_userInterfaces/ADM_dialog/Q_encoding.cpp

```diff
@@ -220,6 +228,21 @@ void DIA_encodingQt4::setPhasis(const char *n)
         WRITEM(labelPhasis,n);
     }
 }
+
+/**
+    \fn setFileName(const char *n)
+*/
+void DIA_encodingQt4::setFileName(const char *n)
+{
+    outputFileName = n;    // dont copy string, the pointer is fine.
```

Comment:
Yeah, probably better safe than sorry.

Reply:
Thanks, the former implementation worked because `std::string::c_str() `– if I am not mistaken – currently returns the pointer to the underlying data, and the `std::string logFileName` member of admSaver is not changed anymore once the class ctor is completed. If any of these circumstances changes, the pointer most likely would be invalid.